webui/react/src/hooks/useModal/Checkpoint/useModalCheckpointRegister.tsx

```diff
@@ -188,12 +145,46 @@ const useRegisterCheckpointModal = (onClose?: (checkpointUuid?: string) => void)
   }, []);
 
   const launchNewModelModal = useCallback((state: ModalState) => {
-    const { checkpointUuid } = state;
-    closeModal();
-    onClose?.(checkpointUuid);
-  }, [ closeModal, onClose ]);
+    modalClose();
+    onClose?.(state.checkpointUuid);
+  }, [ modalClose, onClose ]);
+
+  const fetchModels = useCallback(async () => {
+    try {
+      const response = await getModels({
+        archived: false,
+        orderBy: 'ORDER_BY_DESC',
+        sortBy: validateDetApiEnum(
+          V1GetModelsRequestSortBy,
+          V1GetModelsRequestSortBy.LASTUPDATEDTIME,
+        ),
+      }, { signal: canceler.signal });
+      setModels(prev => {
+        if (isEqual(prev, response.models)) return prev;
+        return response.models;
+      });
+    } catch(e) {
+      handleError(e, {
```

Comment:
I've had this question with useModal and antd modals. do we want these crucial calls to actually throw instead? eg
add the information you have to the error and [augment it](https://github.com/determined-ai/saas/blob/49e3a3d9713c2a222f0db857622f8d92521748f2/web/src/utils/error.ts#L20) and let the modal open call throw. either useModal or antd should then handle that as a critical failure

Reply:
also not the goal of this pr so not blocking

Reply:
I may need to chat with you about this async to get a better idea of this approach.